src/frontend/servers/y-provider/src/middlelayers.ts

```diff
@@ -0,0 +1,59 @@
+import { NextFunction, Request, Response } from 'express';
+import * as ws from 'ws';
+
+import {
+  COLLABORATION_SERVER_ORIGIN,
+  COLLABORATION_SERVER_SECRET,
+} from '@/env';
+
+import { logger } from './utils';
+
+export const httpSecurity = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  // Origin check
+  const origin = req.headers['origin'];
+  if (origin && COLLABORATION_SERVER_ORIGIN !== origin) {
+    logger('CORS policy violation: Invalid Origin', origin);
+
+    res
+      .status(403)
+      .json({ error: 'CORS policy violation: Invalid Origin', origin });
+    return;
+  }
+
+  // Secret API Key check
+  const apiKey = req.headers['authorization'];
+  if (apiKey !== COLLABORATION_SERVER_SECRET) {
+    res.status(403).json({ error: 'Forbidden: Invalid API Key' });
+    return;
+  }
+
+  next();
+};
```

Comment:
Wouldn't it be better to reuse an existing solution? Express might already provide tools for Bearer token authentication, which would save us from maintaining this ourselves.  

On a side note, should we clarify the exact type of authorization header we expect, such as Bearer?  

This can be refactored later, as the priority is to implement the feature quickly.

Reply:
Yes, on a next PR.

Reply:
okay I am aligned !